Uses PHP8's constructor property promotion core/Command/App,/Background, and /Broadcast

fsamapoor opened the pull request:
Following https://github.com/nextcloud/server/pull/38636, https://github.com/nextcloud/server/pull/38637, and https://github.com/nextcloud/server/pull/38638 PRs, I have made the required adjustments to the `/core/Command/` namespace as well.

I figured I should split the changes into multiple PRs to make reviewing the changes easier.

This PR refactors `/core/Command/App`, `/core/Command/Background`, and `/core/Command/Broadcast` classes by using PHP8's constructor property promotion to remove redundant lines and make the code cleaner.

Commit:
Uses PHP8's constructor property promotion.
 in core/Command/App, /Background, and /Broadcast classes.

Signed-off-by: Faraz Samapoor <fsa@adlas.at>

come-nc reviewed (commented):
Same comment as the other PR regarding code style when only one property, otherwise good to go.

Commit:
Moves single constructor parameters to new lines.

Based on:
https://github.com/nextcloud/server/pull/38764#discussion_r1227630895

Signed-off-by: Faraz Samapoor <fsa@adlas.at>
